### DISCUSSION TITLE
Is Custom-schedule example only for Cooja?

### DISCUSSION THREAD

**Discussion by carolinapinho** [contiki-ng/contiki-ng#1902] [Q&A]
Hello!
I am trying to customize a scheduler for a TSCH network with cc2650 launchpads and i was testing the Custom-schedule example because my objective is to start from this one and build my own scheduler. Now, i want to understand if this example was build only for cooja because i never worked with it and if those "potential neighbor (in Cooja) " refered in the code  have their addresses determined through the _for_ cycles present in the code (Lines 83 to 91)  due to some network creation characteristics (i.e. addresses alocation) of cooja that i dont know of or something like that.
If this is the case, if I want to apply this example directly to my cc2650 devices I will have to make changes to this discovery of neighboring addresses in order to proceed with its scheduling.
I hope my question was clear.
Another question: This code is intended only for the node that will be the coordinator, right? what code do i have to flash on the other nodes?
Thank you so much in advance

**Comment by atiselsts** [comment]
The neighbor discovery depended schedule allocation is not par of the example, so yes, it will only work in Cooja. It's more of an example of how to use the TSCH API for scheduling slotframes and cells, not a complete solution ready for experimentation in real networks.

This PR may be helpful - it shows how to add callback function (`NETSTACK_CONF_ROUTING_NEIGHBOR_ADDED_CALLBACK` / `NETSTACK_CONF_ROUTING_NEIGHBOR_REMOVED_CALLBACK`) when the IPv6 neighbors are changed: https://github.com/contiki-ng/contiki-ng/pull/1520/files

The code in the custom schedule example is intended both for coordinator and for nodes.
